src/rest/response.cpp

```diff
@@ -91,6 +85,16 @@ void Response::SetCallback(void)
     mCallback = true;
 }
 
+void Response::SetContentType(const std::string &aContentType)
+{
+    mHeader[OT_REST_RESPONSE_CONTENT_TYPE_HEADER] = aContentType;
+}
+
+std::string Response::GetContentType() const
+{
+    return mHeader.find(OT_REST_RESPONSE_CONTENT_TYPE_HEADER)->second;
```

Comment:
Why not access by `[]`?
```suggestion
    return mHeader[OT_REST_RESPONSE_CONTENT_TYPE_HEADER];
```

Reply:
That does not work:

```
/home/sag/projects/openthread/ot-br-posix/src/rest/response.cpp: In member function ‘std::string otbr::rest::Response::GetContentType() const’:
/home/sag/projects/openthread/ot-br-posix/src/rest/response.cpp:93:34: error: passing ‘const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >’ as ‘this’ argument discards qualifiers [-fpermissive]
   93 |     return mHeader["Content-Type"];
      |                                  ^
```

Reply:
More detailed explanation: https://stackoverflow.com/a/262872/184207

Maybe `at()` is the better option here as we know it must exist?